### ISSUE TITLE
igraph_all_st_mincuts: implement for undirected graphs

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1364]
`igraph_all_st_mincuts` and `igraph_all_st_cuts` are only implemented for directed graphs. It would be nice to implement them for undirected ones too.

Am I mistaken that converting the undirected graph to a mutually connected directed one is sufficient (though likely more than necessary) for this?

Possibly related: #204

**Comment by vtraag** [comment]
In the original publication, [Provan and Shier](https://dx.doi.org/10.1007/s004539900020) describe in section 3.1 a particular approach for st-cuts in undirected graphs. It would probably be best to implement that. Most of the groundwork is already there for the st-cuts in directed graphs anyway (their framework is reasonably general).

It is probably possible to to convert an undirected graph to a mutually connected directed graph as a temporary shortcut. In terms of the partition(s) that you end up with, that approach should probably work. In terms of the edges that are cut, there are of course way too many types of cuts being generated. Each undirected edge will offer two possibilities to cut one of the directed edges, and that goes for each undirected edge. You then end up with a number of cuts that is exponential in the number of edges in the cut in the undirected graph.

**Comment by szhorvat** [comment]
> In the original publication, [Provan and Shier](https://dx.doi.org/10.1007/BF01961544) describe in section 3.1 a particular approach for st-cuts in undirected graphs. It would probably be best to implement that.

First, let me make it clear that I agree with this.

> Each undirected edge will offer two possibilities to cut one of the directed edges, and that goes for each undirected edge.

But I do not think this is true. This function does not find arbitrary cuts. It only finds s-t cuts. The order of s-t (not t-s) defines the direction. Therefore, no two directed cuts will be equivalent in the undirected sense.

Example on a small random graph:

![image](https://user-images.githubusercontent.com/1212871/78132415-ffef4a80-741c-11ea-983b-c8ce092b32ac.png)

I wanted to support undirected graphs with this simple hack in the Mathematica interface, but I need to make absolutely sure that I am not making a mistake when thinking that this is correct ... I've gotten things like this wrong too often in the past by missing some special cases.

**Comment by vtraag** [comment]
Yes, @szhorvat, I believe you are correct. I didn't yet think of the fact that the cut itself is directed of course. That should work as a work around.
